src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -276,7 +276,7 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
 
                             var fileInfo = new FileInfo(output.FullName);
                             Task copyTask = session.EventStream.CopyToAsync(fs);
-                            Task shouldExitTask = copyTask.ContinueWith((task) => shouldExit.Set());
+                            Task shouldExitTask = copyTask.ContinueWith((task) => shouldExit.Set(), TaskScheduler.Default);
```

Comment:
I wonder if the result of `copyTask` should be observed, just so we have an exception when `CopyToAsync` fails.

Reply:
Or could be as simple as `copyTask.ContinueWith(x => x.Exception, TaskContinuationOptions.OnlyOnFaulted);`